core/trino-main/src/main/java/io/trino/execution/QueryInfo.java

```diff
@@ -438,4 +445,35 @@ public String toString()
                 .add("fieldNames", fieldNames)
                 .toString();
     }
+
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE
+                + queryId.getRetainedSizeInBytes()
+                + session.getRetainedSizeInBytes()
+                + MoreSizeOf.sizeOf(self)
+                + estimatedSizeOf(fieldNames, SizeOf::estimatedSizeOf)
+                + estimatedSizeOf(query)
+                + sizeOf(preparedQuery, SizeOf::estimatedSizeOf)
+                + queryStats.getRetainedSizeInBytes()
+                + sizeOf(setCatalog, SizeOf::estimatedSizeOf)
+                + sizeOf(setSchema, SizeOf::estimatedSizeOf)
+                + sizeOf(setPath, SizeOf::estimatedSizeOf)
+                + sizeOf(setAuthorizationUser, SizeOf::estimatedSizeOf)
+                + estimatedSizeOf(setSessionProperties, SizeOf::estimatedSizeOf, SizeOf::estimatedSizeOf)
+                + estimatedSizeOf(resetSessionProperties, SizeOf::estimatedSizeOf)
+                + estimatedSizeOf(setRoles, SizeOf::estimatedSizeOf, SelectedRole::getRetainedSizeInBytes)
+                + estimatedSizeOf(addedPreparedStatements, SizeOf::estimatedSizeOf, SizeOf::estimatedSizeOf)
+                + estimatedSizeOf(deallocatedPreparedStatements, SizeOf::estimatedSizeOf)
+                + sizeOf(startedTransactionId, TransactionId::getRetainedSizeInBytes)
+                + estimatedSizeOf(updateType)
+                + sizeOf(outputStage, StageInfo::getRetainedSizeInBytes)
+                + estimatedSizeOf(referencedTables, TableInfo::getRetainedSizeInBytes)
+                + estimatedSizeOf(routines, RoutineInfo::getRetainedSizeInBytes)
+                + failureInfo.getRetainedSizeInBytes()
+                + estimatedSizeOf(referencedTables, TableInfo::getRetainedSizeInBytes)
+                + estimatedSizeOf(inputs, Input::getRetainedSizeInBytes)
+                + sizeOf(output, Output::getRetainedSizeInBytes)
+                + sizeOf(resourceGroupId, ResourceGroupId::getRetainedSizeInBytes);
```

Comment:
i wish there was a way to test this. 
it should be possible with static (byte) code analysis, except for those comments

(just a thought, no change requested)